cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -201,13 +211,26 @@ private synchronized OutlinePlugin.ErrorCode startTunnel(
       }
     }
 
-    final ShadowsocksConfig proxyConfig = config.proxy;
+    final shadowsocks.Config configCopy = new shadowsocks.Config();
+    configCopy.setHost(config.proxy.host);
+    configCopy.setPort(config.proxy.port);
+    configCopy.setCipherName(config.proxy.method);
+    configCopy.setPassword(config.proxy.password);
+    configCopy.setPrefix(config.proxy.prefix);
+    final shadowsocks.Client client;
+    try {
+      client = new shadowsocks.Client(configCopy);
```

Comment:
It would be helpful to provide a Client factory that takes a JSON object instead (perhaps proto in the future), so the Android implementation doesn't have to know about the fields.

Reply:
Agreed.  However, given the large change required in each platform's code, I don't think that should go in this PR.